src/request-sender.ts

```diff
@@ -93,10 +105,31 @@ export default class RequestSender {
     }
 
     private _prependHost(url: string): string {
-        if (!this._options || !this._options.host || /^https?:\/\//.test(url)) {
+        if (!this._options.host || /^https?:\/\//.test(url)) {
             return url;
         }
 
         return `${this._options.host.replace(/\/$/, '')}/${url.replace(/^\//, '')}`;
     }
+
+    private _shouldCacheRequest(options: RequestOptions) {
```

Comment:
Probably a good practice to specify the return type in the method signature. i.e.: `_shouldCacheRequest(options: RequestOptions): boolean`.